### PR TITLE
KEYCLOAK-5679

### PR DESCRIPTION
**Pull request by staszekbp**
IE in compatibility mode does not add text/html to Accept header

### CODE REVIEW

**Comment by stianst** [comment]
Please squash to a single commit

**Comment by staszekbp** [comment]
Ok, I squashed to a single commit.

File: adapters/oidc/spring-security/src/test/java/org/keycloak/adapters/springsecurity/authentication/HttpHeaderInspectingApiRequestMatcherTest.java [hunk 1]
```diff
@@ -40,22 +40,11 @@ public void setUp() throws Exception {
     }
 
     @Test
-    public void testMatches() throws Exception {
-        assertTrue(apiRequestMatcher.matches(request));
-    }
-
-    @Test
-    public void testMatchesBrowserRequest() throws Exception {
```
Comment (sebastienblanc) [inline adapters/oidc/spring-security/src/test/java/org/keycloak/adapters/springsecurity/authentication/HttpHeaderInspectingApiRequestMatcherTest.java]: We still need a test for browser requests

File: adapters/oidc/spring-security/src/test/java/org/keycloak/adapters/springsecurity/authentication/KeycloakAuthenticationEntryPointTest.java [hunk 2]
```diff
@@ -83,14 +83,6 @@ public void testCommenceWithRedirectNotRootContext() throws Exception {
     }
 
     @Test
-    public void testCommenceWithUnauthorizedWithAccept() throws Exception {
-        request.addHeader(HttpHeaders.ACCEPT, "application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8");
-        authenticationEntryPoint.commence(request, response, null);
-        assertEquals(HttpStatus.UNAUTHORIZED.value(), response.getStatus());
-        assertNotNull(response.getHeader(HttpHeaders.WWW_AUTHENTICATE));
-    }
```
Comment (sebastienblanc) [inline adapters/oidc/spring-security/src/test/java/org/keycloak/adapters/springsecurity/authentication/KeycloakAuthenticationEntryPointTest.java]: We still need to test this.

**Comment by staszekbp** [comment]
Ok, I restored these tests with some modifications which were needed because of my changes in HttpHeaderInspectingApiRequestMatcher class.

**Comment by sebastienblanc** [comment]
@stianst This is good to go

**Pull request merged**